utilities/ner_indexes/ner_indexes.py

```diff
@@ -0,0 +1,126 @@
+import re
+import csv
+import sys
+import json
+import string
+import difflib
+import argparse
+import unicodedata
+from unidecode import unidecode
+from gensim.parsing.preprocessing import preprocess_string, strip_tags, strip_multiple_whitespaces, strip_punctuation
+
+
+def preprocess_primary_name(name):
+    name = re.sub(r'\s\(.*\)', '', name)
+    return name
+
+
+# Used to skip labels that have non-Latin characters
+def check_latin_chars(s):
+    for ch in s:
+        if ch.isalpha():
+            if 'LATIN' not in unicodedata.name(ch):
+                return False
+    return True
+
+
+def preprocess_text(text):
+    custom_filters = [lambda x: x.lower(), strip_multiple_whitespaces]
+    return unidecode(' '.join(preprocess_string(text, custom_filters)))
+
+
+def get_all_names(record):
+    ror_id = record['id']
+    primary_name = preprocess_primary_name(record['name'])
+    aliases = record['aliases']
+    labels = [label['label'] for label in record.get('labels', [])]
+    names = [preprocess_text(name) for name in [
+        primary_name] + aliases + labels if check_latin_chars(name) and len(name) > 5]
+    return names
+
+
+def create_names_dict(data_dump_file):
+    names_dict = {}
+    with open(data_dump_file, 'r+') as f_in:
+        records = json.load(f_in)
+        for record in records:
+            ror_id = record['id']
+            names_dict[ror_id] = get_all_names(record)
+    return names_dict
+
+
+def find_start_stop_index(substring, string):
+    substring_list = preprocess_text(substring).split()
```

Comment:
I think `preprocess_text` was already run when names were added to the dictionary, is it needed to run it again here?

Reply:
Removed.